cpp/src/parquet/file_reader.h

```diff
@@ -146,6 +148,11 @@ class PARQUET_EXPORT ParquetFileReader {
   /// WARNING: The returned PageIndexReader must not outlive the ParquetFileReader.
   std::shared_ptr<PageIndexReader> GetPageIndexReader();
 
+  /// Returns the BloomFilterReader. Only one instance is ever created.
+  ///
+  /// WARNING: The returned BloomFilterReader must not outlive the ParquetFileReader.
```

Comment:
Should we return a raw pointer (or maybe const reference) instead of a `shared_ptr`?  Would it be more clear that the lifetime of the `BloomFilterReader` is bound to the lifetime of the `ParquetFileReader`.

Reply:
Good idea, let me try it

Reply:
I'd prefer returning a `const BloomFilterReader*` or `BloomFilterReader*` so we can have an optimization to return a nullptr when we are sure none of bloom filter exists.